numpy/lib/tests/test_shape_base.py

```diff
@@ -27,6 +27,27 @@ def test_3d(self):
         assert_array_equal(apply_along_axis(np.sum, 0, a),
                            [[27, 30, 33], [36, 39, 42], [45, 48, 51]])
 
+    def test_preserve_subclass(self):
+        m = np.matrix(np.ones((4, 3)))
+        result = apply_along_axis(np.sum, 0, m)
```

Comment:
For this test case, it is not obvious to me that the output should in fact be `np.matrix`.